src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -61,38 +65,59 @@ public class Swerve extends SubsystemBase implements Logged {
 
     private SwerveDrivePoseEstimator poseEstimator;
 
+    private SwerveSetpointGenerator setpointGenerator;
+
+    private SwerveSetpoint swerveSetpoint;
+
+    public static ChassisSpeeds desiredChassisSpeeds = new ChassisSpeeds();
+
+    protected final static SwerveKinematicLimits KINEMATIC_LIMITS = new SwerveKinematicLimits();
+    static {
+        KINEMATIC_LIMITS.MAX_DRIVE_VELOCITY = AutoConstants.MAX_SPEED_METERS_PER_SECOND; // m/s
+        KINEMATIC_LIMITS.MAX_DRIVE_ACCELERATION = 30.0; // m/s^2
```

Comment:
this needs to be tuned, but i have no idea what it should be

Reply:
Yes.. I don't know how we might calculate that, I'll try to do some more research later...